### PR TITLE
build(.gitignore): ignore `pyrightconfig.json` and `poetry.toml`

### PR DESCRIPTION
**Pull request by gbaian10**
## Description

I referred to the `.gitignore` template from this [webpage](https://www.toptal.com/developers/gitignore/api/python) and suggest adding two items to be ignored:

1. `pyrightconfig.json` – For users who use VS Code, they might have their own preferred configurations for `pyright/pylance`.

2. `poetry.toml` – This repository uses Poetry to manage packages. This file can be used to configure some Poetry-related environment variables for the project, such as creating the virtual environment inside the project directory instead of under the .cache folder in the home directory.

## Checklist

- [ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly
  - [ ] Check if there are any broken links in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -113,3 +113,10 @@ venv.bak/
 
 # ruff
 .ruff_cache
+
+# LSP config files
+pyrightconfig.json
+
+### Python Patch ###
+# Poetry local configuration file - https://python-poetry.org/docs/configuration/#local-configuration
+poetry.toml
```
Comment (Lee-W) [inline .gitignore:122]: I'm kinda curious what the behavior you decide but we shouldn't add it to commitizen

File: .gitignore [hunk 2]
```diff
@@ -113,3 +113,10 @@ venv.bak/
 
 # ruff
 .ruff_cache
+
+# LSP config files
+pyrightconfig.json
```
Comment (Lee-W) [inline .gitignore:118]: I would suggest adding it to your `~/.gitignore` instead. I don't like to add things to ignore that's not directly related to commitizen (yep ".vscode" is kinda not my taste either)
